src/log.cpp

```diff
@@ -310,16 +313,29 @@ void Logger::logToOutputs(LogLevel lev, const std::string &combined,
 //// *LogOutput methods
 ////
 
-void FileLogOutput::open(const std::string &filename)
+void FileLogOutput::setFile(const std::string &filename)
 {
-	m_stream.open(filename.c_str(), std::ios::app | std::ios::ate);
+	actionstream << "Log messages are saved to " << filename << std::endl;
+
+	std::ifstream ifile(filename.c_str(), std::ios::binary | std::ios::ate);
+	bool truncate = ifile.tellg() > DEBUGFILE_SIZE_MAX;
+	ifile.close();
+
+	if (!truncate) {
+		m_stream.open(filename.c_str(), std::ios::app | std::ios::ate);
+	} else {
+		actionstream << "The log file grew too big. "
+			"I will remove the old log messages. " << std::endl;
```

Comment:
Active voice seems out of place here as `I` is very ambiguous. It'd be much more pleasing to deliver the same message in a more passive tone. e.g. `The log file grew too big. The older log messages will be removed.`

Reply:
I've changed it to passive.